macros/secondary_calculations/secondary_calculation_period_over_period.sql

```diff
@@ -11,25 +11,40 @@
     {%- endset-%}
     
     {%- if calc_config.comparison_strategy == 'difference' -%}
-        {% do return (adapter.dispatch('metric_comparison_strategy_difference', 'metrics')(metric_name, calc_sql)) %}
+        {% do return (adapter.dispatch('metric_comparison_strategy_difference', 'metrics')(metric_name, calc_sql, metric_config_dict)) %}
     
     {%- elif calc_config.comparison_strategy == 'ratio' -%}
-        {% do return (adapter.dispatch('metric_comparison_strategy_ratio', 'metrics')(metric_name, calc_sql)) %}
+        {% do return (adapter.dispatch('metric_comparison_strategy_ratio', 'metrics')(metric_name, calc_sql, metric_config_dict)) %}
     
     {-% else -%}
         {% do exceptions.raise_compiler_error("Bad comparison_strategy for period_over_period: " ~ calc_config.comparison_strategy ~ ". calc_config: " ~ calc_config) %}
     {%- endif -%}
 
 {% endmacro %}
 
-{% macro default__metric_comparison_strategy_difference(metric_name, calc_sql) -%}
-    coalesce({{ metric_name }}, 0) - coalesce(
+{% macro default__metric_comparison_strategy_difference(metric_name, calc_sql, metric_config_dict) -%}
+    {%- if metric_config_dict.get("default_value_null", False) %}
```

Comment:
I only tangentially saw the discussion about what to call this config, so if it's already been through five rounds of feedback then feel free to ignore. But: 

This doesn't feel intuitive to know what it does. Is it setting default values to null? Is it the value we provide instead in the event of a null? 

I wonder if it should be `treat_null_as_zero` or something (or its opposite, `are_nulls_meaningful`)? 

This is only for the coalesce behaviour in secondary calculations yeah?

Reply:
the only feedback on naming I got here was that `fill_missing_values` was bad, which I wholeheartedly agree with. 

Really not sure what the right way to represent what's happening here is! We are overriding the outputted values for metrics without observations from 0 (default) to null. Can't really imagine a world where you'd want a value here other than null or 0, so I lean towards keeping it a boolean rather than a user supplying some arbitrary value. 

Naming is tough! @callum-mcdata @jtcohen6 what are your thoughts here?

Reply:
I'm a little less bothered on what to name this because I want to ensure that every single option that can be used with metrics is clearly defined in our metrics page. That being said, I agree that an easily understood name is best.

Definitely aligned on it being a boolean value. Replacing nulls with other values would potentially break some loose contracts we have with consumers around the value always being a numeric.

Naming ideas:
- `null_values_treated_as_zero`
- `null_observations_treated_as_zero`

I kind of like something that may be a bit too long but is **reallly** on the nose about what we're doing

Reply:
> * `null_values_treated_as_zero`

I'd phrase this in the imperative voice: `treat_null_values_as_zero`. Then it's clearly an instruction to the metrics package.